src/ui/components/Timeline/Timeline.tsx

```diff
@@ -103,30 +116,42 @@ export default function Timeline() {
     }
   };
 
-  const onMouseUp = (event: MouseEvent) => {
```

Comment:
This event handler was attached too low in the DOM before. If you moused-out and then released, this code would never execute.